internal/server/event_handlers.go

```diff
@@ -325,7 +325,7 @@ func (s *EventServer) eventMatchesAlert(ctx context.Context, event *eventv1.Even
 }
 
 // cleanupMetadata removes metadata entries which are not used for alerting
-func cleanupMetadata(event *eventv1.Event) {
+func cleanupMetadata(event *eventv1.Event) map[string]string {
```

Comment:
```suggestion
func cleanupMetadata(event eventv1.Event) map[string]string {
```
Since the event isn't mutated in the func, anymore, it's better to not make it a pointer.

Reply:
done!